src/webapp/reports/autogenerated-forms/DataEntryItem.tsx

```diff
@@ -92,34 +95,123 @@ function checkVisibleRelatedDataElement(
+        }
+        case "TEXT": {
+            const booleanFromTextValue = value !== "Not sure";
+            return rule.condition === String(booleanFromTextValue);
+        }
+        default:
+            return rule.condition === String(value);
+    }
+}
+
+export function applyNumericComparison(rule: { condition: string }, value: Value) {
+    const [operator, comparisonValue = ""] = rule.condition.split(" ");
+    const numericalValue = parseInt(value as string);
+    const numericalComparisonValue = parseInt(comparisonValue);
+
+    switch (operator) {
+        case ">":
+            return numericalValue > numericalComparisonValue;
+        case "<":
+            return numericalValue < numericalComparisonValue;
+        case ">=":
+            return numericalValue >= numericalComparisonValue;
+        case "<=":
+            return numericalValue <= numericalComparisonValue;
+        case "==":
+        case "===":
+            return numericalValue === numericalComparisonValue;
+        case "!=":
+        case "!==":
+            return numericalValue !== numericalComparisonValue;
+        default:
+            throw new Error(`Unsupported operator: ${operator}`);
+    }
 }
 
 export function useApplyRules(props: UseApplyRulesProps): UseApplyRulesReturn {
     const { dataElement, dataFormInfo, period } = props;
-    if (dataElement.rules.length === 0) return { isDisabled: false, isVisible: true };
-    const disabledRule = dataElement.rules.find(rule => rule.type === "disabled");
-    const disabledValue = disabledRule ? getValueAndVerifyCondition(disabledRule, dataFormInfo, period) : false;
-    const visibleValue = checkVisibleRule({ dataElement, dataFormInfo, period });
+    const totalRules =
+        dataFormInfo.metadata.dataForm.totalRules?.filter(rule => rule.dataElements.includes(dataElement.id)) ?? [];
+    if (dataElement.rules.length === 0 && totalRules.length === 0) return { isDisabled: false, isVisible: true };
+
+    const disabledValue = checkDisabledRule({ dataElement, dataFormInfo, period });
+    const visibleValue = checkVisibleRule({ dataElement, dataFormInfo, period, totalRules });
+
     return { isDisabled: disabledValue, isVisible: visibleValue };
 }
 
-export function checkVisibleRule(options: UseApplyRulesProps) {
-    const { dataElement, dataFormInfo, period } = options;
+export function checkVisibleRule(options: UseApplyRulesProps): boolean {
+    const { dataElement, dataFormInfo, period, totalRules } = options;
     const visibleRule = dataElement.rules.find(rule => rule.type === "visible");
-    const visibleValue = visibleRule ? getValueAndVerifyCondition(visibleRule, dataFormInfo, period) : true;
-    return visibleValue;
+    const visibleDataElementTotalRule = totalRules?.find(rule => rule.type === "visible");
+
+    return visibleRule || visibleDataElementTotalRule
+        ? getValueAndVerifyCondition(visibleRule, dataFormInfo, period, visibleDataElementTotalRule) ?? true
```

Comment:
We can remove the `?? true` since we already are returning that as a default value inside the method